packages/sdk-core/src/model/configuration/BacktraceConfiguration.ts

```diff
@@ -0,0 +1,51 @@
+import { BacktraceAttachment } from '../report/BacktraceAttachment';
+import { BacktraceDatabaseConfiguration } from './BacktraceDatabaseConfiguration';
+
+// Submission information
+export type BacktraceBasicSubmissionInformation = {
+    /**
+     * The server address (submission URL) is required to submit exceptions from your project to your Backtrace instance.
+     *
+     * The Server Address must be in the following format: https://submit.backtrace.io/{subdomain}/{submission-token}/json.
+     */
+    url: string;
+    timeout?: number;
+    ignoreSslCertificate?: boolean;
+};
+export type BacktraceLegacySubmitInformation = BacktraceBasicSubmissionInformation & { token: string };
+export type BacktraceSubmissionInformation = BacktraceBasicSubmissionInformation | BacktraceLegacySubmitInformation;
+
+export type BacktraceMetricsSupport = {
+    metricsSubmissionUrl?: string;
+    enable: boolean;
+    ignoreSslCertificate?: boolean;
+    /**
+     * Indicates how often crash free metrics are sent to Backtrace.
+     * By default, session events are sent on application startup/finish, and every 30 minutes while the game is running.
+     */
+    autoSendInterval?: number;
+};
+
+export type BasicBacktraceClientSettings = BacktraceBasicSubmissionInformation & {
+    /**
+     * Limits the number of reports the client will send per minute. If set to '0', there is no limit.
+     * If set to a value greater than '0' and the value is reached, the client will not send any reports until the next minute.
+     */
+    rateLimit: number;
+    /**
+     * Attributes are additional metadata that can be attached to error and crash reports. You can use attributes to filter,
+     * aggregate, analyze, and debug errors in the Backtrace console.
+     */
+    userAttributes?: Record<string, unknown>;
+    /**
+     * Attachments are additional files/data that can be send with error to Backtrace.
+     */
+    attachments?: BacktraceAttachment[];
+
+    /**
+     * Metrics such as crash free users and crash free sessions
+     */
+    metrics?: BacktraceMetricsSupport;
+};
+
+export type BacktraceConfiguration = BasicBacktraceClientSettings & BacktraceDatabaseConfiguration;
```

Comment:
why are we not using `interface` here in these options (not the union ones)?

Reply:
why should we? Do we want to inherit from them? Is there any better reason to use interface than type?

Reply:
it seems more suitable to use an interface to define an object contract. I would use `type` to define a type alias, like  `type BacktraceConfiguration = BasicBacktraceClientSettings & BacktraceDatabaseConfiguration;`, or `type Key = string | number`.

Reply:
Pushed adjusted code base on our conversation